runtime/compiler/z/codegen/J9TreeEvaluator.cpp

```diff
@@ -3396,6 +3425,664 @@ J9::Z::TreeEvaluator::ArrayCopyBNDCHKEvaluator(TR::Node * node, TR::CodeGenerato
+
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_BNE, node, endLabel); // branch to end if not set
+   generateS390LabelInstruction(cg, TR::InstOpCode::LABEL, node, startLabel);
+     
+   if (isStatic)
+      {
+      // Fills in J9JITWatchedStaticFieldData.fieldClass
+      TR::Register *fieldClassReg = cg->evaluate(isWrite ? getDirectWrtbarJ9ClassNode(node): node->getFirstChild());
+      generateRXInstruction(cg, TR::InstOpCode::getStoreOpCode(), node, fieldClassReg, generateS390MemoryReference(dataBlockReg, (int32_t)(offsetof(J9JITWatchedStaticFieldData, fieldClass)), cg));
+      deps->addPostCondition(fieldClassReg, TR::RealRegister::AssignAny);
+      }
+   TR::ResolvedMethodSymbol *methodSymbol = node->getByteCodeInfo().getCallerIndex() == -1 ? cg->comp()->getMethodSymbol(): cg->comp()->getInlinedResolvedMethodSymbol(node->getByteCodeInfo().getCallerIndex());
+        
+   // setup linkage 
+   TR::Register *cpAddressReg = cg->allocateRegister();
+   TR::Register *cpIndexReg = cg->allocateRegister();
+
+   generateRegLitRefInstruction(cg, TR::InstOpCode::LG, node, cpAddressReg, (uintptrj_t) methodSymbol->getResolvedMethod()->constantPool(), TR_ConstantPool, deps, 0, 0);
+
+   generateRILInstruction(cg, TR::InstOpCode::LGFI, node, cpIndexReg, symRef->getCPIndex());
+   deps->addPostCondition(cpAddressReg, TR::RealRegister::GPR1);
+   deps->addPostCondition(cpIndexReg, TR::RealRegister::GPR2);
+             
+   // Z specific linkage
+   TR::Register * scratchReg1 = cg->allocateRegister();
+   TR::Register * scratchReg2 = cg->allocateRegister();
+
+   deps->addPostCondition(scratchReg1, cg->getEntryPointRegister());
+   deps->addPostCondition(scratchReg2, cg->getReturnAddressRegister());
+                  
+   resultReg = cpIndexReg;
+
+   cg->stopUsingRegister(scratchReg1);
+   cg->stopUsingRegister(scratchReg2);
+   cg->stopUsingRegister(cpAddressReg);
+         
+
+   TR::SymbolReference * helperSymRef = cg->symRefTab()->findOrCreateRuntimeHelper(helperIndex, false, false, false);
+   cg->resetIsLeafMethod();
+
+   TR::Instruction * call = generateDirectCall(cg, node, false /*myself*/, helperSymRef, deps);//do call;
+   call->setNeedsGCMap(0x0000FFFF);
+
+   /*
+   For instance field offset, the result returned by the vmhelper includes header size.
+   subtract the header size to get the offset needed by field watch helpers
+   */
+   if (!isStatic)
+      generateRILInstruction(cg, TR::InstOpCode::getSubtractLogicalImmOpCode(), node, resultReg, static_cast<uint32_t>(TR::Compiler->om.objectHeaderSizeInBytes()));
+
+   generateRXInstruction(cg, TR::InstOpCode::getStoreOpCode(), node, resultReg, generateS390MemoryReference(dataBlockReg, offsetInDataBlock, cg));
+   generateS390BranchInstruction(cg, TR::InstOpCode::BRC, TR::InstOpCode::COND_MASK15, node, endLabel);
```

Comment:
This branch seems to be branching to the very next instruction. Did we miss something here?

Reply:
Must have overlooked this at some point when moving instructions around. Fixed here: https://github.com/eclipse/openj9/pull/4743/commits/d9b2454aa0e8a125f7dff8932ccfb42ccbd82255